mkdocs/docs/services/rhs.md

```diff
@@ -0,0 +1,172 @@
+
+By following these steps, the algorithm generates a Merkle proof of existence or non-existence for a given key in a Sparse Merkle Tree without revealing the specific element being proved.
+
+Example:
+
+```go
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/iden3/go-merkletree-sql/v2"
+	proof "github.com/iden3/merkletree-proof"
+)
+
+func main() {
+	ctx := context.Background()
+	rhsURL := "https://rhs-staging.polygonid.me"
+	state := "e12084d0d72c492c703a2053b371026bceda40afb9089c325652dfd2e5e11223"
+	revocationNonce, _ := merkletree.NewHashFromBigInt(big.NewInt(670966937))
+
+	cli := &proof.HTTPReverseHashCli{URL: rhsURL}
+
+	// get identity state roots (ClT root, ReT root, RoT root)
+	node := getIdentityStateRoots(cli, ctx, state)
+	json, err := json.Marshal(node)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(json))
+
+	// get non-revocation proof by ReT root and revocation nonce
+	proof, err := cli.GenerateProof(ctx, node.Children[1], revocationNonce)
+	if err != nil {
+		panic(err)
+	}
+	jsonProof, err := proof.MarshalJSON()
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(string(jsonProof))
+}
+
+func getIdentityStateRoots(cli *proof.HTTPReverseHashCli, ctx context.Context, state string) proof.Node {
+	s, err := merkletree.NewHashFromHex(state)
+	if err != nil {
+		panic(err)
+	}
+
+	stateValues, err := cli.GetNode(ctx, s)
+	if err != nil {
+		panic(err)
+	}
+
+	return stateValues
+}
+```
+
+### Credential Status
+
+If identity is published to RHS then we can check the credential status by doing a reverse hash lookup for the credential revocation nonce. In this case credential issued by this identity may have credential status type `Iden3ReverseSparseMerkleTreeProof`
+
+```json
+...{    
+"type": "Iden3ReverseSparseMerkleTreeProof"    
+"id": "https://rhs-staging.polygonid.me/node",    
+"revocationNonce": 3409889195,
+}...
+```
+
+***id*** - address of RHS node
+
+***revocationNonce*** - credential revocation nonce
+
+***type*** - type of the credential status
+
+JSON-LD context: https://schema.iden3.io/core/jsonld/iden3proofs.jsonld
+
+Type implementations:
+
+- Go: [https://github.com/iden3/go-schema-processor](https://github.com/iden3/go-schema-processor)
+- JS: [https://github.com/0xPolygonID/js-sdk](https://github.com/0xPolygonID/js-sdk)
+
+### RHS Implementation
+
+[https://github.com/iden3/reverse-hash-service](https://github.com/iden3/reverse-hash-service)
+
+### Library to interact with RHS
+
+[https://github.com/iden3/merkletree-proof](https://github.com/iden3/merkletree-proof)
+
+### Links
+
+[https://docs.iden3.io/publications/pdfs/Merkle-Tree.pdf](https://docs.iden3.io/publications/pdfs/Merkle-Tree.pdf)
+
+[https://github.com/iden3/go-merkletree-sql](https://github.com/iden3/go-merkletree-sql)
+
+[https://github.com/0xPolygonID/js-sdk](https://github.com/0xPolygonID/js-sdk)
```

Comment:
```suggestion
[https://github.com/0xPolygonID/js-sdk/blob/2f711aaa45126bf16781d4daa4fe2f166470ce41/src/credentials/revocation.ts](https://github.com/0xPolygonID/js-sdk)
```

Reply:
I think better to leave a link to the SDK itself.